### PR TITLE
DCS-1736 migration from nomis

### PR DESCRIPTION
**Pull request by michalnawrockidj**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      videoLinkAppointmentRepository.countByLocationIdIsNull(),
+      videoLinkBookingRepository.countByPrisonIdIsNull(),
+    )
+  }
+
+  fun updateVideoLink(videoLinkBooking: VideoLinkBooking) {
+    val mainAppointment = videoLinkBooking.appointments.get(HearingType.MAIN)
+    val perAppointment = videoLinkBooking.appointments.get(HearingType.PRE)
+    val postAppointment = videoLinkBooking.appointments.get(HearingType.POST)
+    if (mainAppointment == null) {
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt:34]: is this actually possible? I didn't think there would be any cases where there wouldn't be a main appointment for a booking in our local DB?

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    videoLinkBookingRepository.findByPrisonIdIsNull(PageRequest.of(0, batchSize)).stream().forEach { v -> updateVideoLink(v) }
+
+    return VideoLinkAppointmentMigrationResponse(
+      videoLinkAppointmentRepository.countByLocationIdIsNull(),
+      videoLinkBookingRepository.countByPrisonIdIsNull(),
+    )
+  }
+
+  fun updateVideoLink(videoLinkBooking: VideoLinkBooking) {
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt:30]: presumably this should run in a transaction?

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    val mainAppointment = videoLinkBooking.appointments.get(HearingType.MAIN)
+    val perAppointment = videoLinkBooking.appointments.get(HearingType.PRE)
+    val postAppointment = videoLinkBooking.appointments.get(HearingType.POST)
+    if (mainAppointment == null) {
+      videoLinkAppointmentRepository.deleteAll(videoLinkBooking.appointments.values)
+      videoLinkBookingRepository.delete(videoLinkBooking)
+    } else {
+      val nomisMainAppointment = updateAppointment(mainAppointment)
+      if (nomisMainAppointment == null) {
+        videoLinkAppointmentRepository.deleteAll(videoLinkBooking.appointments.values)
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt:40]: does cascade remove mean that this isn't necessary? can we not just delete the booking?

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt (hunk 1 above)
Comment (michalnawrockidj) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt:38]: Programmatically it is possible get allow to give me null back  so I can just do nothing with this booking by covering all logic by mainAppointment?.let {...} or implement the logic delete this booking.

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt (hunk 2 above)
Comment (michalnawrockidj) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt:33]: yes, sorted

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt (hunk 3 above)
Comment (michalnawrockidj) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/VideoLinkBookingMigrationService.kt]: simplified and confirmed by integration test

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**